### ISSUE TITLE
Style/GuardClause causes excessively long lines

### ISSUE THREAD

**Issue by ekohl** [rubocop/rubocop#10708]
## Is your feature request related to a problem? Please describe.

While I appreciate guard clauses on short lines, I think they reduce readability when they're too long. IMHO `Style/GuardClause` should at least consider the resulting line length.

## Describe the solution you'd like

The Style/GuardClause gets a maximum resulting line length parameter. If the resulting line is longer than x characters, the cop does nothing. This gives authors a choice between disabling the cop and keeping the cop for short lines.

## Describe alternatives you've considered

I have not considered alternatives. That does not mean there are none and I'm welcoming other suggestions.

## Additional context

Taking the first 2 examples from https://docs.rubocop.org/rubocop/1.29/cops_style.html#styleguardclause it can be argued that they're easier to read and thus understand the code:
```ruby
# bad
def test
  if something
    work
  end
end

# good
def test
  return unless something

  work
end
```

However, taking another example:
```ruby
# Real world example, keeping indenting for full effect
      unless @resource[:ports].include?(:unset)
        ports_protos = Array(@resource[:ports]).select { |x| x['port'].nil? }.map { |x| x['protocol'] }
      end

# Gets converted into:
      ports_protos = Array(@resource[:ports]).select { |x| x['port'].nil? }.map { |x| x['protocol'] } unless @resource[:ports].include?(:unset)
```

To me it is now harder to effectively follow the flow of the code. Note that the `unless` keyword is at character 102, which means I need to scan a full line to see. On smaller screens it may even be invisible by default.

**Comment by tejasbubane** [comment]
@ekohl [`Style/GuardClause` does check for line length](https://github.com/rubocop/rubocop/blob/40e83efe122490e81df96c7034302e92d6285c7c/lib/rubocop/cop/style/guard_clause.rb#L161) configured under [`Layout/LineLength` cop as `Max`](https://github.com/rubocop/rubocop/blob/40e83efe122490e81df96c7034302e92d6285c7c/lib/rubocop/cop/autocorrect_logic.rb#L92). [The default is 120](https://github.com/rubocop/rubocop/blob/40e83efe122490e81df96c7034302e92d6285c7c/config/default.yml#L997).

**Comment by ekohl** [comment]
Interesting. I still feel like the cop generates unreadable code in many cases while still generating better code in some. Perhaps this is because we disable the LineLength cop? How would I then deal with this?

**Comment by jonas054** [comment]
I think you should enable `Layout/LineLength`. It's not just guard clauses that are hard to read when they're too long.

**Comment by dvandersluis** [comment]
I agree, `Style/GuardClause` shouldn't be concerned with the length of a line, there are other cops for that.

**Comment by ekohl** [comment]
Sounds like my resolution is to disable `Style/GuardClause` then.
